src/lib/Libtpp/tpp_util.c

```diff
@@ -2081,18 +2081,19 @@ tpp_get_addresses(char *names, int *count)
 tpp_addr_t *
 tpp_get_local_host(int sock)
 {
-	struct sockaddr addr;
+	struct sockaddr_storage addrs;
+	struct sockaddr *addr = (struct sockaddr *)&addrs;
```

Comment:
is this really ok? it's usually very dangerous to cast pointer of one type to another, unless you are casting a void pointer, It can lead to unexpected runtime issues.

Reply:
It should be fine, according to [spec](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/socket.h.html)

> When a sockaddr_storage structure is cast as a sockaddr structure, the ss_family field of the sockaddr_storage structure shall map onto the sa_family field of the sockaddr structure. When a sockaddr_storage structure is cast as a protocol-specific address structure, the ss_family field shall map onto a field of that structure that is of type sa_family_t and that identifies the protocol's address family.

Reply:
Thanks Vincent. From what I understand, this states that it's ok to cast sockaddr_storage to sockaddr. My concern is regarding the pointer cast from sockaddr_storage * to sockaddr *. I've been bitten in the past by casting int * to long *, the compiler did weird things with it at higher optimization levels. So, just to be on the safe side, can we do "struct sockaddr addr = (struct sockaddr ) addrs;" instead ?

Reply:
@agrawalravi90 Its ok to cast pointer also in this case, if you read spec link given by @vstumpf then it says that:
> The <sys/socket.h> header shall define the sockaddr_storage structure. This structure shall be:
>
>    Large enough to accommodate all supported protocol-specific address structures
>
>    Aligned at an appropriate boundary so that pointers to it can be cast as pointers to protocol-specific address structures and used to access the fields of those structures without alignment problems

Reply:
Thanks for looking into it, I guess if the spec guarantees the safety then we can use it.